### PR TITLE
Fix Label3D, TextMesh & Font not following project default theme in editor

### LINKED ISSUE #85693: Label3D font does not follow project default theme in editor anymore (in-game is fine)
### Godot version

v4.2.stable.official [46dc27791]

### System information

Godot v4.2.stable - Windows 10.0.22621 - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 4070 Ti (NVIDIA; 31.0.15.3667) - AMD Ryzen 9 7900X 12-Core Processor (24 Threads)

### Issue description

See screenshot.

In 4.1.3, Label3D shows custom theme font in editor.

In 4.2, Label3D shows a default sans-serif font in editor.

image

### Steps to reproduce

* Start the project using 4.1.3
* Make a theme.
* Customize the theme's default font
* Make your theme the project default (must enable advanced settings to reach this setting)
* Add Label3D to scene
* Clone the project folder
* Open the clone in 4.2 (triggering the 4.1.3-to-4.2 one-way conversion on the project)

Result: Label3D will (probably) not show your custom font in the 4.2 editor.

### Minimal reproduction project

[Label3DFontBugMRP.zip](https://github.com/godotengine/godot/files/13537687/Label3DFontBugMRP.zip)

### PR DESCRIPTION
**Pull request by Mickeon**
_Seems_ to fix https://github.com/godotengine/godot/issues/85693 without side effects, and I honestly have no idea why.

### CODE REVIEW

File: scene/3d/label_3d.cpp [hunk 1]
```diff
@@ -791,12 +791,17 @@ Ref<Font> Label3D::_get_font_or_default() const {
...
-	StringName theme_name = "font";
+	const StringName theme_name = "font";
 	List<StringName> theme_types;
 	ThemeDB::get_singleton()->get_native_type_dependencies(get_class_name(), &theme_types);
 
 	ThemeContext *global_context = ThemeDB::get_singleton()->get_default_theme_context();
-	for (const Ref<Theme> &theme : global_context->get_themes()) {
+	List<Ref<Theme>> themes = global_context->get_themes();
+	if (Engine::get_singleton()->is_editor_hint()) {
+		themes.push_front(ThemeDB::get_singleton()->get_project_theme()); // Preview in the editor properly.
```
Comment (Chubercik) [inline scene/3d/label_3d.cpp:801]: >// Preview in the editor properly.

Not sure if this comment is necessary

Comment (Mickeon) [reply scene/3d/label_3d.cpp:801]: To _me_ it feels weird that it has to be done this way. I can remove the comment though.

Comment (Chubercik) [reply scene/3d/label_3d.cpp:801]: I'm content either way

**Review by bruvzg — COMMENTED** [review]
Exactly the same code is used in the `TextMesh` and by `Font` to get the default font, so these might need the same change, but I have no idea if this is a correct way to fix it.

**Comment by Mickeon** [comment]
Curious, but you're right. The problem *does* also exist for those two.
PR before,
image
PR after,
image

**Review by Chubercik — APPROVED** [review]

**Review by KoBeWi — APPROVED** [review]
The reason why it works is that the global theme context in editor does not contain project theme:
https://github.com/godotengine/godot/blob/0ace0a129284ffc6646b199699c1607a316fcec0/scene/theme/theme_db.cpp#L278-L280
You are pushing it to front of the iterated theme list, so the Label3D checks it and finds the font (`get_font()` will always return something if a default font is assigned).

btw the `get_fallback_theme()` code below is most likely dead, as fallback theme is the same as the last theme in the List above.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**